packages/protocol/src/utils/type-util.ts

```diff
@@ -184,3 +184,13 @@ export function hasArrayProp(object: AnyObject, propertyKey: string, optional =
     const property = (object as any)[propertyKey];
     return property !== undefined ? Array.isArray(property) : optional;
 }
+
+/**
+ * Validates whether the given object has a property with the given key.
+ * @param object The object that should be validated
+ * @param propertyKeys The keys of the properties
+ * @returns `true` if the object has property with matching key.
+ */
+export function hasProperty<T extends AnyObject>(object: AnyObject, ...propertyKeys: (keyof T)[]): boolean {
+    return propertyKeys.every(propertyKey => propertyKey in object);
```

Comment:
After applying my suggestion for `SearchAutocompleteSuggestion.is` this utility function is unused.

As we all know, developers tend to be lazy and might use this function in typeguards to simply check against all  properties of the interface/type (not accounting for optional properties).
I fear that providing these function might result in incomplete type guard checks in the future (which can be easily overlooked in reviews).
I would rather opt for removing this and going for explicit checks per property.
( We already have the `hasStringProp(), hasNumberProp()...` utilities for that)